### PR TITLE
jUnit & Hamcrest task

### PR DESCRIPTION
**Pull request by NataliaBodrug**
Hello,
Please review jUnit and Hamcrest task.
Thank you!

### CODE REVIEW

File: src/test/java/jUnit/CalculateNumbers.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public static void main(String[] args) {
+        getSumOfNumbers(2, 3);
+        subtractNumbers(5, 3);
+        multiplyNumbers(2, 3);
+        divideNumbers(6, 2);
+    }
+
+    public static int getSumOfNumbers(int n, int m) {
+        int sum = n + m;
+        return sum;
```
Comment (am-bee73) [inline src/test/java/jUnit/CalculateNumbers.java:14]: you may inline all in return statement

File: src/test/java/jUnit/CalculateNumbers.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public static int subtractNumbers(int n, int m) {
+        int difference = n - m;
+        return difference;
```
Comment (am-bee73) [inline src/test/java/jUnit/CalculateNumbers.java:19]: same for all methods in this class

File: src/test/java/jUnit/tests/CalculateNumbersTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        int sum = CalculateNumbers.getSumOfNumbers(numbers.getN(), numbers.getM());
+
+        assertEquals(String.format("Actual sum %d not corresponding expected %d", sum, 9), 9, sum);
+    }
+
+    @Test
+    public void testSubstractNumbers() {
+        int difference = CalculateNumbers.subtractNumbers(numbers.getN(), numbers.getM());
+
+        assertThat("Result of substraction is correct", difference, is(3));
```
Comment (am-bee73) [inline src/test/java/jUnit/tests/CalculateNumbersTest.java:43]: in order to remove hardcoded values you may add an expected variable like:
actual =  CalculateNumbers.subtractNumbers(numbers.getN(), numbers.getM());
expected = numbers.getN() - numbers.getM();

and compare this two, assertThat(actual, is(expected))

File: src/test/java/jUnit/tests/CalculateNumbersTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Test
+    public void testMultiplyNumbers() {
+        int multiplication = CalculateNumbers.multiplyNumbers(numbers.getN(), numbers.getM());
+
+        assertThat("Result of multiplication is correct", multiplication, is(18));
```
Comment (am-bee73) [inline src/test/java/jUnit/tests/CalculateNumbersTest.java:50]: same here and to the method below

File: src/test/java/jUnit/tests/CalculateNumbersTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    public void setTestAfter() {
+        printOnScreen();
+        System.out.println("Test finished successfully.\n");
+    }
+
+    @Test
+    public void testGetSumOfNumbers() {
+        int sum = CalculateNumbers.getSumOfNumbers(numbers.getN(), numbers.getM());
+
+        assertEquals(String.format("Actual sum %d not corresponding expected %d", sum, 9), 9, sum);
```
Comment (am-bee73) [inline src/test/java/jUnit/tests/CalculateNumbersTest.java:36]: also here

File: src/test/java/jUnit/tests/CalculateNumbersTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        numbers.setN(6);
+        numbers.setM(3);
+        System.out.println("The setted numbers are: " + numbers.toString());
+        System.out.println("Test data setted in @Before.");
+
+    }
+
+    @After
+    public void setTestAfter() {
+        printOnScreen();
```
Comment (am-bee73) [inline src/test/java/jUnit/tests/CalculateNumbersTest.java:28]: do you really need here also the message from println & from method ?

File: src/test/java/jUnit/CalculateNumbers.java (hunk 1 above)
Comment (NataliaBodrug) [reply src/test/java/jUnit/CalculateNumbers.java:14]: removing int sum = n+m , and writing return n+m?

File: src/test/java/jUnit/CalculateNumbers.java (hunk 2 above)
Comment (NataliaBodrug) [reply src/test/java/jUnit/CalculateNumbers.java:19]: ok

File: src/test/java/jUnit/tests/CalculateNumbersTest.java (hunk 6 above)
Comment (NataliaBodrug) [reply src/test/java/jUnit/tests/CalculateNumbersTest.java:28]: just to see that this is possible way

File: src/test/java/jUnit/CalculateNumbers.java (hunk 1 above)
Comment (NataliaBodrug) [reply src/test/java/jUnit/CalculateNumbers.java]: done

File: src/test/java/jUnit/tests/CalculateNumbersTest.java (hunk 3 above)
Comment (NataliaBodrug) [reply src/test/java/jUnit/tests/CalculateNumbersTest.java]: done

File: projectClasa.iml [hunk 7]
```diff
@@ -11,5 +11,7 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
```
Comment (am-bee73) [inline projectClasa.iml:13]: better add .iml to gitignore

Comment (NataliaBodrug) [reply projectClasa.iml:13]: i have it

Comment (am-bee73) [reply projectClasa.iml:13]: if its there it shouldn't appear here

**Pull request merged**